cpp/src/cylon/net/ops/gather.cpp

```diff
@@ -153,6 +154,67 @@ cylon::Status cylon::mpi::Gather(const std::shared_ptr<cylon::TableSerializer> &
   return cylon::Status::OK();
 }
 
+cylon::Status cylon::mpi::GatherArrowBuffer(const std::shared_ptr<arrow::Buffer> &buf,
+                                            int gather_root,
+                                            const std::shared_ptr<cylon::CylonContext> &ctx,
+                                            std::vector<std::shared_ptr<arrow::Buffer>> &buffers) {
+
+  std::vector<int32_t> all_buffer_sizes;
+  if (AmIRoot(gather_root, ctx)) {
+    all_buffer_sizes.resize(ctx->GetWorldSize(), 0);
+  }
+
+  int32_t size = static_cast<int32_t>(buf->size());
+  int status = MPI_Gather(&size,
+                          1,
+                          MPI_INT32_T,
+                          all_buffer_sizes.data(),
+                          1,
+                          MPI_INT32_T,
+                          gather_root,
+                          MPI_COMM_WORLD);
+  if (status != MPI_SUCCESS) {
+    return cylon::Status(cylon::Code::ExecutionError, "MPI_Gather failed when receiving buffer sizes!");
+  }
+
+  std::shared_ptr<arrow::Buffer> all_buf = std::move(arrow::AllocateBuffer(0).MoveValueUnsafe());
+  std::vector<int32_t> disps;
+
+  if (AmIRoot(gather_root, ctx)) {
+    auto total_size = std::accumulate(all_buffer_sizes.begin(), all_buffer_sizes.end(), 0);
+    all_buf = std::move(arrow::AllocateBuffer(total_size).MoveValueUnsafe());
+
+    disps.resize(ctx->GetWorldSize(), 0);
+    for (int i = 1; i < all_buffer_sizes.size(); ++i) {
+      disps[i] = disps[i-1] + all_buffer_sizes[i-1];
+    }
+  }
+
+  status = MPI_Gatherv(buf->data(),
```

Comment:
```suggestion
  status = MPI_Gatherv(reinterpret_cast<const void*>(buf->data()),
```
let's cast this. I think it generates a stupid warning otherwise. :disappointed:

Reply:
this cast does not compile.

Reply:
ah never mind then